probe-rs/src/architecture/arm/core/armv7m.rs

```diff
@@ -637,6 +637,13 @@ impl<'probe> Armv7m<'probe> {
             sequence,
         })
     }
+
+    fn set_current_core_status(&mut self, status: CoreStatus) {
```

Comment:
Having to implement this for every arch is what bothers me. I wonder if we could make this more generic.

Can't we have this function as a normal function in the `architecture/arm/core` module and it takes a reference to `Memory`. `State` and the `status` as arguments?

Reply:
I've added `update_core_status` to `arch/arm/core/mod.rs` that does this, and made the Core implementations delegate to that. Definitely a good idea to keep all of that in once place.

Reply:
Do we need the delegate? Otherwise I like this solution a lot :)

I rather clutter internal code than trait interfaces, which is why I am kind of trying to get rid of this additional method that does not really fit the trait :)

Reply:
This specific one isn't a trait fn, so I don't think I understand your comment. Also see my comment below